sdcm/remote.py

```diff
@@ -885,24 +885,30 @@ def run(self, cmd, timeout=None, ignore_status=False,
         :param watch_stdout_pattern: Mark a timestamp (time.time()) on which
                 the given stdout pattern appeared first. This will be
                 available on result.stdout_pattern_found_at
+        :param retry: unconditionally retry times if execution raise exception.
         :return: avocado.utils.process.CmdResult object with the result of
                 the remote command executed.
         """
-        if args is None:
-            args = ()
-        if verbose:
-            self.log.debug("Running '%s'", cmd)
+        for i in range(retry + 1):
+            try:
+                if args is None:
+                    args = ()
+                if verbose:
+                    self.log.debug("Running '%s'", cmd)
```

Comment:
I think the args checking and log print can be outside of the loop.

Reply:
How about keeping the debug statement insider the loop, then we know which command is retried?

Reply:
no problem